src/main/kotlin/blackjack/model/GameResult.kt

```diff
@@ -1,7 +1,15 @@
 package blackjack.model
 
-enum class GameResult(val label: String) {
-    WIN("Win"),
-    LOSE("Lose"),
-    DRAW("Draw")
-}
\ No newline at end of file
+enum class GameResult(
+    private val multiplier: Double,
+) {
+    BLACKJACK_WIN(1.5),
+    WIN(1.0),
+    DRAW(0.0),
+    LOSE(-1.0),
```

Comment:
much better 😉 👍 
also u can try like this

```suggestion
enum class GameResult(val calculateEarnings: (Int) -> Int) {
    BLACKJACK_WIN({ bet -> (bet * 1.5).toInt() }),
    WIN({ bet -> bet }),
    DRAW({ _ -> 0 }),
    LOSE({ bet -> -bet });
}
```

Reply:
And, `GameResult` also can use the `BetAmount` 😉

Reply:
Thank you I took your suggestion, I replaced the multiplier with function-based strategy in GameResult, and also updated it to use the Bet type directly :)